internal/services/databricks/databricks_virtual_network_peerings_resource.go

```diff
@@ -0,0 +1,326 @@
+		if err != nil {
+			return fmt.Errorf("unable to parse 'workspace_id': %+v", err)
+		}
+
+		id = vnetpeering.NewVirtualNetworkPeeringID(subscriptionId, d.Get("resource_group_name").(string), workspaceId.WorkspaceName, d.Get("name").(string))
+
+		existing, err := client.Get(ctx, id)
+		if err != nil {
+			if !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing Databricks %s: %s", id, err)
+			}
+		}
+
+		if !response.WasNotFound(existing.HttpResponse) {
+			return tf.ImportAsExistsError("azurerm_databricks_virtual_network_peering", id.ID())
+		}
+	} else {
+		// For import and update, you need to parse the id else you will have an empty subscription,
+		// resource group name and workspace name in the PUT call causing a 400 Bad Request...
+		rawId, err := vnetpeering.ParseVirtualNetworkPeeringID(d.Id())
+		if err != nil {
+			return fmt.Errorf("unable to parse 'id': %+v", err)
+		}
+
+		id = *rawId
+	}
+
+	peer := vnetpeering.VirtualNetworkPeering{
+		Name:       &id.VirtualNetworkPeeringName,
+		Properties: expandDatabricksVirtualNetworkPeeringProperties(d),
+	}
+
+	// The RP always creates the same vNet ID for the Databricks internal vNet in the below format:
+	// '/subscriptions/{subscription}/resourceGroups/{group1}/providers/Microsoft.Network/virtualNetworks/workers-vnet'
+	fmtString := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/workers-vnet"
+	virtualNetworkId := fmt.Sprintf(fmtString, id.SubscriptionId, id.ResourceGroupName)
+
+	peer.Properties.DatabricksVirtualNetwork = &vnetpeering.VirtualNetworkPeeringPropertiesFormatDatabricksVirtualNetwork{
+		Id: utils.String(virtualNetworkId),
+	}
+
+	peerMutex.Lock(resourceLock)
+	defer peerMutex.Unlock(resourceLock)
+
+	if err := pluginsdk.Retry(300*time.Second, retryDatabricksVnetPeeringsClientCreateUpdate(d, id, peer, meta)); err != nil {
+		return err
+	}
+
+	d.SetId(id.ID())
+
+	return resourceDatabricksVirtualNetworkPeeringRead(d, meta)
+}
+
+func resourceDatabricksVirtualNetworkPeeringRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataBricks.VnetPeeringsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// for the import scenario, once the peering is created I can generate the workspace ID from the peering ID
+	// NOTE: this may not be right because I don't know if it is valid to create a peering with a
+	// workspace across subscription boundaries?
+	workspaceId := vnetpeering.NewWorkspaceID(id.SubscriptionId, id.ResourceGroupName, id.WorkspaceName)
+
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on Databricks %s: %+v", *id, err)
```

Comment:
fwiw other resources use:

```suggestion
		return fmt.Errorf("retrieving %s: %+v", *id, err)
```

Reply:
Fixed.